### PR TITLE
[ROCm] Rocm62 fixes

### PR DESCRIPTION
**Pull request by Ruturaj4**
1) Interactive causes bazel to output more useful info when running locally.
2) Work around quirk with rocm version when it ends with 0.
3) Ubu22 and higher have a package name conflict between the debian versions and the AMD provided versions.
4) Use env variable to set compiler type.

### CODE REVIEW

**Comment by Ruturaj4** [comment]
@superbobry Could you please take a look?

I will squash the commits once approved.

**Comment by dfm** [comment]
@superbobry is OOO, but I'm happy to take a look. It all looks good to me - can you please squash!

**Comment by Ruturaj4** [comment]
@dfm Done thanks!

**Review by dfm — APPROVED** [review]

File: build/rocm/Dockerfile.ms [hunk 1]
```diff
@@ -59,6 +59,7 @@ FROM rocm_base AS rt_build
 ARG JAX_VERSION
 ARG JAX_COMMIT
 ARG XLA_COMMIT
+ARG JAX_USE_CLANG
```
Comment (mrodden) [inline build/rocm/Dockerfile.ms:62]: I think you can drop this still

**Review by zahiqbal — COMMENTED** [review]
looks good to me

**Review by dfm — APPROVED** [review]

**Comment by dfm** [comment]
@Ruturaj4 — I'm sorry to say that there were unrelated issues with the builds yesterday. Could you rebase onto main (don't forget to squash!) one more time then we can merge? Thank you!

**Comment by Ruturaj4** [comment]
@dfm no worries, at all! thanks so much!

**Review by dfm — APPROVED** [review]

**Pull request merged**